File: pegasusX/apps/supplier-app-android/app/src/test/java/com/pegasusx/supplier/PlanningTests.kt
package com.pegasusx.supplier

import com.pegasusx.supplier.data.model.ForecastConfidence
import com.pegasusx.supplier.util.brainForecastLine
import com.pegasusx.supplier.util.factoryPlanningDisabledCode
import com.pegasusx.supplier.util.isForecastBlocked
import com.pegasusx.supplier.util.planBrainTabFromQuery
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Assert.assertTrue
import org.junit.Test

class PlanningTests {
    @Test
    fun testPlanBrainTabAndBlockedForecast() {
        assertEquals("brain", planBrainTabFromQuery("brain"))
        assertEquals("planning", planBrainTabFromQuery(null))
        val blocked = ForecastConfidence(
            blockedReason = "sparsity_blocked",
            label = "insufficient_history",
        )
        assertTrue(isForecastBlocked(blocked))
        assertNull(brainForecastLine(blocked, listOf(1.0, 2.0, 3.0)))
        assertEquals(
            "factory_planning_disabled",
            factoryPlanningDisabledCode(409, """{"error":"factory_planning_disabled"}"""),
        )
    }
}

File: pegasusX/apps/supplier-app-ios/SupplierAppTests/PlanningTests.swift
import XCTest
@testable import SupplierAppIOS

final class PlanningTests: XCTestCase {
    func testPlanBrainTabAndBlockedForecast() {
        XCTAssertEqual(planBrainTabFromQuery("brain"), .brain)
        XCTAssertEqual(planBrainTabFromQuery(nil), .planning)
        let blocked = ForecastConfidence(
            lowUnits: 10,
            highUnits: 20,
            confidencePct: 80,
            baselineSource: nil,
            blockedReason: "sparsity_blocked",
            label: "insufficient_history"
        )
        XCTAssertTrue(blocked.isBlocked)
        XCTAssertNil(brainForecastLine(confidence: blocked, accuracyPoints: [1, 2, 3]))
        XCTAssertEqual(factoryPlanningDisabledCode(status: 409, body: "{\"error\":\"factory_planning_disabled\"}"), "factory_planning_disabled")
    }
}
